src/documentation/io/built-in-hadoop.md

```diff
@@ -0,0 +1,156 @@
+---
+layout: default
+title: "Apache Hadoop InputFormat IO"
+permalink: /documentation/io/built-in/hadoop/
+---
+
+[Pipeline I/O Table of Contents]({{site.baseurl}}/documentation/io/io-toc/)
+
+# Hadoop InputFormat IO
+
+A HadoopInputFormatIO is a Transform for reading data from any source which
+implements Hadoop InputFormat. For example- Cassandra, Elasticsearch, HBase, Redis, Postgres, etc.
+
+HadoopInputFormatIO has to make several performance trade-offs in connecting to InputFormat, so if there is another Beam IO Transform specifically for connecting to your data source of choice, we would recommend using that one, but this IO Transform allows you to connect to many data sources that do not yet have a Beam IO Transform.
+
+You will need to pass a Hadoop Configuration with parameters specifying how the read will occur. Many properties of the Configuration are optional, and some are required for certain InputFormat classes, but the following properties must be set for all InputFormats:
+
+mapreduce.job.inputformat.class: The InputFormat class used to connect to your data source of choice.
+key.class: The key class returned by the InputFormat in 'mapreduce.job.inputformat.class'.
+value.class: The value class returned by the InputFormat in 'mapreduce.job.inputformat.class'.
+
```

Comment:
* `mapreduce.job.inputformat.class` - The `InputFormat` class used to connect to your data source of choice.
* `key.class` - The `Key` class returned by the `InputFormat` in `mapreduce.job.inputformat.class`.
* `value.class` -  The `Value` class returned by the `InputFormat` in `mapreduce.job.inputformat.class`.